apps/coordinator/src/components/Wallet/WalletPendingTransactions.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useEffect, useState } from "react";
+import { useSelector } from "react-redux";
+import { Box, Paper } from "@mui/material";
+import { styled } from "@mui/material/styles";
+import { useGetClient } from "../../hooks";
+// import { TransactionAnalyzer } from "@caravan/bitcoin";
+import { AnalyzedTransaction, RootState, UTXO } from "./fee-bumping/types";
+import { calculateTimeElapsed } from "./fee-bumping/utils";
+import TransactionTable from "./fee-bumping/PendingTransactionTable";
+import RBFOptionsDialog from "./fee-bumping/rbf/RBFOptionsDialog";
+import AccelerateFeeDialog from "./fee-bumping/rbf/AccelerateFeeDialog";
+import CancelTransactionDialog from "./fee-bumping/rbf/CancelTransactionDialog";
+
+const StyledPaper = styled(Paper)(({ theme }) => ({
+  padding: theme.spacing(3),
+  marginBottom: theme.spacing(2),
+}));
+
+const WalletPendingTransactions: React.FC = () => {
+  const [pendingTransactions, setPendingTransactions] = useState<
+    AnalyzedTransaction[]
+  >([]);
+  const [selectedTx, setSelectedTx] = useState<AnalyzedTransaction | null>(
+    null,
+  );
+  const [showRBFOptions, setShowRBFOptions] = useState(false);
+  const [showIncreaseFees, setShowIncreaseFees] = useState(false);
+  const [showCancelTx, setShowCancelTx] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+
+  const network = useSelector((state: RootState) => state.settings.network);
+  const walletSlices = useSelector((state: RootState) => [
+    ...Object.values(state.wallet.deposits.nodes),
+    ...Object.values(state.wallet.change.nodes),
+  ]);
+  const blockchainClient = useGetClient();
+
+  useEffect(() => {
+    fetchPendingTransactions();
+  }, [network, walletSlices, blockchainClient]);
+
+  const fetchPendingTransactions = async () => {
+    try {
+      const pendingTxs = walletSlices
+        .flatMap((slice) => slice.utxos)
+        .filter((utxo) => !utxo.confirmed);
+
+      const currentNetworkFeeRate = await getCurrentNetworkFeeRate();
+
+      const analyzedTransactions = await Promise.all(
+        pendingTxs.map(async (utxo) =>
+          analyzeTransaction(utxo, currentNetworkFeeRate),
+        ),
+      );
+
+      setPendingTransactions(analyzedTransactions);
+    } catch (error) {
+      console.error("Error fetching pending transactions:", error);
+      setError("Failed to fetch pending transactions. Please try again later.");
+    } finally {
+      setIsLoading(false);
+    }
+  };
```

Comment:
I wonder if this could all be pulled out as a hook, something like `useGetPendingTransactions`. We'd need some way to set an error from inside the hook though.